### PR TITLE
AccountInfo and ContractInfo Parity

### PR DESCRIPTION
**Pull request by danielakhterov**
- Implement `AccountInfo.getCost()`
    - Implement `ContractInfo.getCost()`
    - Update `AccountDeleteTransaction.setDeleteAccountId()`
      to automatically create a `TransactionId` with the correct
      `AccountId`

Signed-off-by: Daniel Akhterov <daniel@launchbadge.com>

### CODE REVIEW

**Comment by mehcode** [comment]
> Update `AccountDeleteTransaction.setDeleteAccountId()`
  to automatically create a `TransactionId` with the correct
  `AccountId`

Please remove that. I'm reasonably certain this _should not_ be done and is instead a bug in the network that we shouldn't patch over.

**Comment by abonander** [comment]
We should at least document it then.

**Comment by mehcode** [comment]
Agreed on that for sure.

File: examples/delete-account.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    let transactionReceipt = await transactionId.getReceipt(client);
+    const newAccountId = transactionReceipt.getAccountId();
+
+    console.log(`account = ${newAccountId}`);
+    console.log("Deleting created account");
+
+    transactionId = await new AccountDeleteTransaction()
+        .setDeleteAccountId(newAccountId)
+        .setTransferAccountId("0.0.3")
+        .setTransactionId(new TransactionId(newAccountId))
```
Comment (abonander) [inline examples/delete-account.js]: I would add a comment here explaining why this is necessary.

File: src/account/AccountInfoQuery.ts [hunk 2]
```diff
@@ -43,6 +44,15 @@ export class AccountInfoQuery extends QueryBuilder<AccountInfo> {
         return this;
     }
 
+    public async getCost(client: BaseClient): Promise<Hbar> {
+        // deleted accounts return a COST_ANSWER of zero which triggers `INSUFFICIENT_TX_FEE`
+        // if you set that as the query payment; 25 tinybar seems to be the minimum to get
```
Comment (abonander) [inline src/account/AccountInfoQuery.ts]: It's not necessarily the "minimum" for `AccountInfoQuery`, it's just probably *enough*.

File: src/contract/ContractInfoQuery.ts [hunk 3]
```diff
@@ -38,6 +40,15 @@ export class ContractInfoQuery extends QueryBuilder<ContractInfo> {
         return this;
     }
 
+    public async getCost(client: BaseClient): Promise<Hbar> {
+        // deleted contracts return a COST_ANSWER of zero which triggers `INSUFFICIENT_TX_FEE`
+        // if you set that as the query payment; 25 tinybar seems to be the minimum to get
```
Comment (abonander) [inline src/contract/ContractInfoQuery.ts]: Same wording here, "the minimum" -> "enough"

File: src/account/AccountDeleteTransaction.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { CryptoService } from "../generated/CryptoService_pb_service";
 import UnaryMethodDefinition = grpc.UnaryMethodDefinition;
 import { CryptoDeleteTransactionBody } from "../generated/CryptoDelete_pb";
 import { AccountId, AccountIdLike } from "./AccountId";
+import { TransactionId } from "../TransactionId";
```
Comment (abonander) [inline src/account/AccountDeleteTransaction.ts]: Unused import.

**Pull request merged**